### PR TITLE
Provide the location when creating glif work

### PR DESCRIPTION
**Pull request by rsheeter**
Provide a set of Path:DesignSpaceLocation(s) to the create IR function. Intended to leave us in a good position to actually build glyph IR.

### CODE REVIEW

File: ufo2fontir/src/source.rs [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
 };
 use log::debug;
 use norad::designspace::{self, DesignSpaceDocument};
 
 pub struct DesignSpaceIrSource {
     designspace_file: PathBuf,
+    designspace_dir: PathBuf,
     ir_paths: Paths,
+    // A cache of locations, valid provided no font info file changes
+    glif_locations: (FileStateSet, HashMap<PathBuf, Vec<DesignSpaceLocation>>),
```
Comment (anthrotype) [inline ufo2fontir/src/source.rs:21]: why would a change in UFO fontinfo.plist invalidate the cache of glyph locations? 
oh maybe I misunderstood, seems like you used the name `font_info` (the return value of font_info_files) to mean a FileStateSet that includes not just source ufos' fontinfo.plist (which I think are irrelevant to that glif_locations cache) but also the designspace file itself (which defines those locations and thus is relevant).
Perhaps use a different name that can't be misunderstood this way

Comment (rsheeter) [reply ufo2fontir/src/source.rs:21]: Naming is the hardest part. Will update. 

My rationale was that font_info.plist includes information that seemed to warrant a global rebuild on change, e.g. upem. I'll add a comment as well as renaming.

Comment (rsheeter) [reply ufo2fontir/src/source.rs:21]: Did that help?

Comment (anthrotype) [reply ufo2fontir/src/source.rs:21]: yes, thanks

File: ufo2fontir/src/source.rs [hunk 2]
```diff
@@ -110,43 +130,107 @@ impl Source for DesignSpaceIrSource {
...
+    ) -> Result<Vec<Box<dyn Work<()>>>, Error> {
+        // A single glif could be used by many source blocks that use the same layer
+        // *gasp*
+        // So resolve each file to 1..N locations in designspace
+
+        let mut work: Vec<Box<dyn Work<()>>> = Vec::new();
+
+        // Do we have the location of glifs written down?
+        // TODO: consider just recomputing here instead of failing
+        if input.font_info != self.glif_locations.0 {
```
Comment (anthrotype) [inline ufo2fontir/src/source.rs:176]: I find a bit opaque the use of an unnamed tuple for this cache of glif locations, when you access things with self.glif_locations.0 and self.glif_locations.1 I wonder what that means. Why do those two have to go together as a tuple, can't they be distinct field of DesignSpaceIrSource struct?
or maybe it's the name of the self.glif_locations that I find confusing; it is appropriate for its .1 item (the hashmap of glif locations), but perhaps not for the tuple as a whole..

Comment (rsheeter) [reply ufo2fontir/src/source.rs]: Clearer?

Comment (anthrotype) [reply ufo2fontir/src/source.rs]: yes thanks

**Review by anthrotype — APPROVED** [review]

**Pull request merged**